validator/client/src/main/java/tech/pegasys/teku/validator/client/duties/BlockProductionDuty.java

```diff
@@ -54,6 +56,14 @@ public BlockProductionDuty(
   public SafeFuture<DutyResult> performDuty() {
     LOG.trace("Creating block for validator {} at slot {}", validator.getPublicKey(), slot);
     return forkProvider.getForkInfo(slot).thenCompose(this::produceBlock);
+    // TODO do we need to call prepareDuty if it has not been called before?
+  }
+
+  @Override
+  public SafeFuture<Void> prepareDuty() {
+    executionPayloadRequestId =
+        UInt64.fromLongBits(SecureRandomProvider.publicSecureRandom().nextLong());
+    return validatorApiChannel.prepareExecutionPayload(slot, executionPayloadRequestId);
```

Comment:
I would move this field initialization to the constructor and make it final 
Does it make sense using `random()` for generating `requestId` here? Wouldn't a simple counter be enough?

Reply:
the idea was to avoid any potential concurrency issues (can they happen here?) and generate different id in case of duty prepared twice (late consensus block arrival). In the latter I assumed we need to refresh the `requestId` (going to change it to `payloadId` as per latest HackMD update).

Reply:
Oh, I follow the logic now... `prepareDuty()` is designed to be called more than once

Reply:
I would suggest using `AtomicInteger` here even if it's a static field to generate rather than random. There's no reason we need to be using a secure random at this point.

I do wonder if we should be using the same requestId when we resend prepareDuty though - otherwise we're leaving the previous "prepare" call hanging and it will never be cancelled or followed by "get".  Might be a question for the API spec as much as anything.

Reply:
I moved to `AtomicLong` and made `executionPayloadId` constant at every preparation